src/pipelines/data_gen.py

```diff
@@ -658,11 +665,13 @@ def gen_observations(all_clusters, lookup_order, document_data, features, parame
     mem_net_features['questions'] = questions
     mem_net_features['input_masks'] = input_masks
     mem_net_features['answers'] = labels
-
+    
+    ids = ["C" + str(i[0]) + "_P" + str(i[1]) for i in zip(clusterids,postids)] 
```

Comment:
fyi you can use tuple unpacking here instead:
`ids = ["C" + str(clusterid]) + "_P" + str(postid) for clusterid, postid in zip(clusterids,postids)]`